router/utils/utils.go

```diff
@@ -1,6 +1,7 @@
 package utils
```

Comment:
the package name is not very indicative of its contents, and in general, util/misc names should be avoided.

We can have a `type` package for `JobParameters`.
and introduce a new `drainer` package for `NewDrainer`